ci: Windows builds fail intermittently

No idea why.

```
npm ERR! gyp: C:\Users\runneradmin\.electron-gyp\22.0.3\common.gypi not found (cwd: D:\a\bob-wallet\bob-wallet\node_modules\bcrypto) while reading includes of binding.gyp while trying to load binding.gyp
```

example: https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019

<details>
<summary>full error </summary>

```
> Run `npm i`

npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: mini-create-react-context@0.[4](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:5).1
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"18.2.0" from the root project
npm WARN   10 more (@cfaester/enzyme-adapter-react-18, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14.0 || ^1[5](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:6).0.0 || ^1[6](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:7).0.0 || ^1[7](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:8).0.0" from mini-create-react-context@0.4.1
npm WARN node_modules/react-router/node_modules/mini-create-react-context
npm WARN   mini-create-react-context@"^0.4.0" from react-router@5.3.3
npm WARN   node_modules/react-router
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0" from mini-create-react-context@0.4.1
npm WARN   node_modules/react-router/node_modules/mini-create-react-context
npm WARN     mini-create-react-context@"^0.4.0" from react-router@5.3.3
npm WARN     node_modules/react-router

> bob-wallet@1.0.0 postinstall
> electron-builder install-app-deps

  • electron-builder  version=23.6.0
  • loaded configuration  file=package.json ("build" field)
  • rebuilding native dependencies  dependencies=bcrypto@5.4.0, bdb@1.4.0, goosig@0.10.0, leveldown@5.6.0, mrmr@0.1.10, node-hid@2.1.2-hsd.0 platform=win32 arch=x64
  ⨯ cannot execute  cause=exit status 1
                    errorOut=npm ERR! code 1
    npm ERR! path D:\a\bob-wallet\bob-wallet\node_modules\bcrypto
    npm ERR! command failed
    npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node-gyp rebuild
    npm ERR! gyp info it worked if it ends with ok
    npm ERR! gyp info using node-gyp@9.1.0
    npm ERR! gyp info using node@16.19.0 | win32 | x64
    npm ERR! gyp info find Python using Python version 3.9.13 found at "C:\hostedtoolcache\windows\Python\3.9.13\x64\python3.exe"
    npm ERR! gyp http GET https://electronjs.org/headers/v22.0.3/node-v22.0.3-headers.tar.gz
    npm ERR! gyp http 200 https://artifacts.electronjs.org/headers/v22.0.3/node-v22.0.3-headers.tar.gz?force_headers_dist=1
    npm ERR! gyp http GET https://electronjs.org/headers/v22.0.3/SHASUMS256.txt
    npm ERR! gyp http GET https://electronjs.org/headers/v22.0.3/win-x[8](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:9)6/node.lib
    npm ERR! gyp http GET https://electronjs.org/headers/v22.0.3/win-x64/node.lib
    npm ERR! gyp http GET https://electronjs.org/headers/v22.0.3/win-arm64/node.lib
    npm ERR! gyp http 200 https://artifacts.electronjs.org/headers/v22.0.3/SHASUMS256.txt?force_headers_dist=1
    npm ERR! gyp http 200 https://artifacts.electronjs.org/headers/v22.0.3/win-x64/node.lib?force_headers_dist=1
    npm ERR! gyp http 200 https://artifacts.electronjs.org/headers/v22.0.3/win-x86/node.lib?force_headers_dist=1
    npm ERR! gyp http 200 https://artifacts.electronjs.org/headers/v22.0.3/win-arm64/node.lib?force_headers_dist=1
    npm ERR! gyp info find VS using VS2022 (17.4.33213.308) found at:
    npm ERR! gyp info find VS "C:\Program Files\Microsoft Visual Studio\2022\Enterprise"
    npm ERR! gyp info find VS run with --verbose for detailed information
    npm ERR! gyp info spawn C:\hostedtoolcache\windows\Python\3.[9](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:10).13\x64\python3.exe
    npm ERR! gyp info spawn args [
    npm ERR! gyp info spawn args   'C:\\hostedtoolcache\\windows\\node\\16.19.0\\x64\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
    npm ERR! gyp info spawn args   'binding.gyp',
    npm ERR! gyp info spawn args   '-f',
    npm ERR! gyp info spawn args   'msvs',
    npm ERR! gyp info spawn args   '-I',
    npm ERR! gyp info spawn args   'D:\\a\\bob-wallet\\bob-wallet\\node_modules\\bcrypto\\build\\config.gypi',
    npm ERR! gyp info spawn args   '-I',
    npm ERR! gyp info spawn args   'C:\\hostedtoolcache\\windows\\node\\16.19.0\\x64\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
    npm ERR! gyp info spawn args   '-I',
    npm ERR! gyp info spawn args   'C:\\Users\\runneradmin\\.electron-gyp\\22.0.3\\common.gypi',
    npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
    npm ERR! gyp info spawn args   '-Dvisibility=default',
    npm ERR! gyp info spawn args   '-Dnode_root_dir=C:\\Users\\runneradmin\\.electron-gyp\\22.0.3',
    npm ERR! gyp info spawn args   '-Dnode_gyp_dir=C:\\hostedtoolcache\\windows\\node\\16.19.0\\x64\\node_modules\\npm\\node_modules\\node-gyp',
    npm ERR! gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\runneradmin\\\\.electron-gyp\\\\22.0.3\\\\<(target_arch)\\\\node.lib',
    npm ERR! gyp info spawn args   '-Dmodule_root_dir=D:\\a\\bob-wallet\\bob-wallet\\node_modules\\bcrypto',
    npm ERR! gyp info spawn args   '-Dnode_engine=v8',
    npm ERR! gyp info spawn args   '--depth=.',
    npm ERR! gyp info spawn args   '--no-parallel',
    npm ERR! gyp info spawn args   '--generator-output',
    npm ERR! gyp info spawn args   'D:\\a\\bob-wallet\\bob-wallet\\node_modules\\bcrypto\\build',
    npm ERR! gyp info spawn args   '-Goutput_dir=.'
    npm ERR! gyp info spawn args ]
    npm ERR! gyp: C:\Users\runneradmin\.electron-gyp\22.0.3\common.gypi not found (cwd: D:\a\bob-wallet\bob-wallet\node_modules\bcrypto) while reading includes of binding.gyp while trying to load binding.gyp
    npm ERR! gyp ERR! configure error 
    npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
    npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (C:\hostedtoolcache\windows\node\16.19.0\x64\node_modules\npm\node_modules\node-gyp\lib\configure.js:284:16)
    npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
    npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)
    npm ERR! gyp ERR! System Windows_NT [10](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:11).0.20348
    npm ERR! gyp ERR! command "C:\\hostedtoolcache\\windows\\node\\[16](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:17).19.0\\x64\\node.exe" "C:\\hostedtoolcache\\windows\\node\\16.19.0\\x64\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
    npm ERR! gyp ERR! cwd D:\a\bob-wallet\bob-wallet\node_modules\bcrypto
    npm ERR! gyp ERR! node -v v16.19.0
    npm ERR! gyp ERR! node-gyp -v v9.1.0
    npm ERR! gyp ERR! not ok
    
    npm ERR! A complete log of this run can be found in:
    npm ERR!     C:\npm\cache\_logs\2023-01-30T06_35_35_859Z-debug-0.log
    
                    command='C:\hostedtoolcache\windows\node\16.19.0\x64\node.exe' 'C:\hostedtoolcache\windows\node\16.19.0\x64\node_modules\npm\bin\npm-cli.js' rebuild bcrypto@5.4.0 bdb@1.4.0 goosig@0.10.0 leveldown@5.6.0 mrmr@0.1.10 node-hid@2.1.2-hsd.0
                    workingDir=
npm ERR! code 1
npm ERR! path D:\a\bob-wallet\bob-wallet
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c electron-builder install-app-deps

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\2023-01-30T06_32_29_9[17](https://github.com/kyokan/bob-wallet/actions/runs/4040964849/jobs/6947611019#step:5:18)Z-debug-0.log
Error: Process completed with exit code 1.
```

</details>

A couple of references:
```
https://github.com/AceCentre/EyeCommander/issues/51
https://forums.meteor.com/t/gyp-error-while-installing-bcrypt-on-windows/54643
https://github.com/serialport/node-serialport/issues/2322

https://github.com/nodejs/node-gyp/issues/2683
https://github.com/Seo-Rii/electron-acrylic-window/issues/84
https://github.com/nodejs/node-gyp/issues/2584
https://github.com/nodejs/node-gyp/issues/2714
```

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -16,7 +16,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [windows-latest, ubuntu-latest]
-        node-version: ['16']
+        node-version: ['18']
 
         # OS-specific commands/variables
         include:

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -29,10 +29,10 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+      - name: Use Node.js
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
 

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -16,7 +16,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [windows-latest, ubuntu-latest]
-        node-version: ['18']
+        node-version: ['16', '18']
 
         # OS-specific commands/variables
         include:

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -36,6 +36,16 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
+      # https://github.com/actions/setup-node/issues/280#issuecomment-1139455898
+      - name: Update node-gyp (windows only)
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          $WhereNode = Get-Command node | Select-Object -ExpandProperty Definition
+          $NodeDirPath = Split-Path $WhereNode -Parent
+          $NodeModulesPath = $NodeDirPath + "\node_modules\npm\node_modules\@npmcli\run-script"
+          cd $NodeModulesPath
+          npm install node-gyp@latest
+
       - name: Install dependencies (linux only)
         if: matrix.os == 'ubuntu-latest'
         run: sudo apt update && sudo apt install libudev-dev libusb-1.0-0-dev -y

--- a/.github/workflows/package.yml
+++ b/.github/workflows/package.yml
@@ -30,10 +30,10 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Use Node.js 16.x
+      - name: Use Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 18
           cache: "npm"
 
       - name: Install dependencies (linux only)

--- a/.github/workflows/package.yml
+++ b/.github/workflows/package.yml
@@ -10,11 +10,11 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-latest, ubuntu-18.04, macos-latest]
+        os: [windows-latest, ubuntu-20.04, macos-latest]
 
         # OS-specific commands/variables
         include:
-          - os: ubuntu-18.04
+          - os: ubuntu-20.04
             cmd_package: npm run package-linux
             out_filename: Bob-linux
           - os: windows-latest

--- a/.github/workflows/package.yml
+++ b/.github/workflows/package.yml
@@ -36,17 +36,22 @@ jobs:
           node-version: 18
           cache: "npm"
 
+      # https://github.com/actions/setup-node/issues/280#issuecomment-1139455898
+      - name: Update node-gyp (windows only)
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          $WhereNode = Get-Command node | Select-Object -ExpandProperty Definition
+          $NodeDirPath = Split-Path $WhereNode -Parent
+          $NodeModulesPath = $NodeDirPath + "\node_modules\npm\node_modules\@npmcli\run-script"
+          cd $NodeModulesPath
+          npm install node-gyp@latest
+
       - name: Install dependencies (linux only)
         if: startsWith(matrix.os, 'ubuntu')
         run: sudo apt update && sudo apt install libudev-dev libusb-1.0-0-dev -y
 
       - name: Install npm packages
-        uses: nick-fields/retry@v2
-        with:
-          timeout_minutes: 10
-          max_attempts: 3
-          retry_on: error
-          command: npm i
+        run: npm i
 
       - name: Install dmg-license (macos only)
         if: startsWith(matrix.os, 'macos')

